sdk/core/System.ClientModel/tests/Pipeline/RequestRetryPolicyTests.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using NUnit.Framework;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using TestHelpers.Internal;
+
+namespace System.ClientModel.Tests;
+
+public class RequestRetryPolicyTests
+{
+    [Test]
+    public void CanRetryErrorResponse()
+    {
+        PipelineOptions options = new();
+        options.Transport = new RetriableTransport("Transport", new int[] { 429, 200 });
```

Comment:
We have a problem at the moment with this approach, that since IEnumerables only go forward, if we need to retry, we can't roll back the stack of policies.